### ISSUE TITLE
[Bug ] Memory leak on latest release 0.2.7

### ISSUE THREAD

**Issue by alimoezzi** [vllm-project/vllm#2624]
I'm able to run TheBloke/dolphin-2.6-mixtral-8x7b-AWQ on 2x 4090s on git hash 1db83e3 but in the new release I receive the following cuda out of memory error:
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/user/projects/repos/vllm/vllm/entrypoints/openai/api_server.py", line 217, in <module>
    engine = AsyncLLMEngine.from_engine_args(engine_args)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/projects/repos/vllm/vllm/engine/async_llm_engine.py", line 617, in from_engine_args
    engine = cls(parallel_config.worker_use_ray,
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/projects/repos/vllm/vllm/engine/async_llm_engine.py", line 321, in __init__
    self.engine = self._init_engine(*args, **kwargs)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/projects/repos/vllm/vllm/engine/async_llm_engine.py", line 366, in _init_engine
    return engine_class(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/projects/repos/vllm/vllm/engine/llm_engine.py", line 112, in __init__
    self._init_cache()
  File "/home/user/projects/repos/vllm/vllm/engine/llm_engine.py", line 339, in _init_cache
    self._run_workers("warm_up_model")
  File "/home/user/projects/repos/vllm/vllm/engine/llm_engine.py", line 977, in _run_workers
    driver_worker_output = getattr(self.driver_worker,
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/projects/repos/vllm/vllm/worker/worker.py", line 143, in warm_up_model
    self.model_runner.capture_model(self.gpu_cache)
  File "/home/user/projects/repos/vllm/.venv/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/projects/repos/vllm/vllm/worker/model_runner.py", line 676, in capture_model
    graph_runner.capture(
  File "/home/user/projects/repos/vllm/vllm/worker/model_runner.py", line 722, in capture
    with torch.cuda.graph(self.graph, pool=memory_pool):
  File "/home/user/projects/repos/vllm/.venv/lib/python3.11/site-packages/torch/cuda/graphs.py", line 197, in __exit__
    self.cuda_graph.capture_end()
  File "/home/user/projects/repos/vllm/.venv/lib/python3.11/site-packages/torch/cuda/graphs.py", line 88, in capture_end
    super().capture_end()
RuntimeError: CUDA error: out of memory
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.
```

Command:
```
RAY_memory_usage_threshold=1 python3 -m vllm.entrypoints.openai.api_server --model=TheBloke/dolphin-2.6-mixtral-8x7b-AWQ  --gpu-memory-utilization 1.0 --tensor-parallel-size 2 --max-model-len 16096
```

**Comment by lroberts7** [comment]
I tried to reproduct the issue with tensor-parallel set to 2 but this encounters the issue I've reported in #2620. 
However, I have a development build with the latest commit and this works fine for me with tensor paralle l set to 1 (I have an 8GPU 80GB A100. So I think this may be specific to hardware the reporter is using:


```bash
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 530.30.02              Driver Version: 530.30.02    CUDA Version: 12.1     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                  Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf            Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA A100-SXM4-80GB           On | 00000000:07:00.0 Off |                    0 |
| N/A   35C    P0               74W / 400W|  80661MiB / 81920MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA A100-SXM4-80GB           On | 00000000:0A:00.0 Off |                    0 |

... nvidia smi output elided ... 

+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A   3877709      C   python3                                   80650MiB |
+---------------------------------------------------------------------------------------+
```

I used the following command:
` RAY_memory_usage_threshold=1 python3 -m vllm.entrypoints.openai.api_server --model=TheBloke/dolphin-2.6-mixtral-8x7b-AWQ --tensor-parallel-size 1 --max-model-len 8192`

so it may also be that the issue lies with the `--gpu-memory-utilization 1.0` flag and execution triggered by that value. I do see this issue when I use that flag but I don't see if I omit `--gpu-memory-utilization 1.0`.

**Comment by lroberts7** [comment]
@alimoezzi actually this works fine for me on tip of main if I instead use `--gpu-memory-utilization 0.9` so maybe this is a weird edge case error? Try using slightly less than 100% of the memory in the device and see if it will run for you.

**Comment by WoosukKwon** [comment]
Should be fixed by #2811
